sklearn/preprocessing/data.py

```diff
@@ -2579,6 +2579,150 @@ def quantile_transform(X, axis=0, n_quantiles=1000,
+    >>> data = [[1, 2], [3, 2], [4, 5]]
+    >>> print(boxcox.fit(data))
+    BoxCoxTransformer(copy=True)
+    >>> print(np.round(boxcox.lambdas_, 3))
+    [ 1.052 -2.345]
+    >>> print(np.round(boxcox.transform(data), 3))
+    [[ 0.     0.342]
+     [ 2.068  0.342]
+     [ 3.135  0.417]]
+
+    Notes
+    -----
+    See also QuantileTransformer(output_distribution='uniform') for another
+    transformer to normalize data to the Gaussian distribution.
+
+    For a comparison of the different scalers, transformers, and normalizers,
+    see :ref:`examples/preprocessing/plot_all_scaling.py
+    <sphx_glr_auto_examples_preprocessing_plot_all_scaling.py>`.
+
+    References
+    ----------
+    G.E.P. Box and D.R. Cox, "An Analysis of Transformations", Journal of the
+    Royal Statistical Society B, 26, 211-252 (1964).
+
+    """
+    def __init__(self, copy=True):
+        self.copy = copy
+
+    def fit(self, X, y=None):
+        """Estimate the optimal lambda for each feature using MLE.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The data used to estimate the Box-Cox transformation parameters.
+
+        y : Ignored
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X = check_array(X, ensure_2d=True, dtype=FLOAT_DTYPES)
+
+        if np.any(X <= 0):
+            raise ValueError("The Box-Cox transformation can only be applied "
+                             "to strictly positive data")
+
+        self.lambdas_ = []
+        for col in X.T:
+            _, lmbda = stats.boxcox(col, lmbda=None)
+            self.lambdas_.append(lmbda)
+        self.lambdas_ = np.array(self.lambdas_)
+
+        return self
+
+    def transform(self, X):
+        """Apply Box-Cox to each feature using the fitted lambdas.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The data to be transformed using the Box-Cox transformation.
+        """
+        check_is_fitted(self, 'lambdas_')
+        X = check_array(X, ensure_2d=True, dtype=FLOAT_DTYPES, copy=self.copy)
+
+        if np.any(X <= 0):
+            raise ValueError("The Box-Cox transformation can only be applied "
+                             "to strictly positive data")
+
+        if not len(self.lambdas_) == X.shape[1]:
+            raise ValueError("Input data has a different number of features "
+                             "than fitting data. Should have {n}, data has {m}"
+                             .format(n=len(self.lambdas_), m=X.shape[1]))
+
+        for i, lmbda in enumerate(self.lambdas_):
+            X[:, i] = stats.boxcox(X[:, i].flatten(), lmbda=lmbda)
+
+        return X
+
+    def inverse_transform(self, X):
+        """Undo the Box-Cox transformation using the fitted lambdas.
+
+        The inverse of the Box-Cox transformation is given by::
+
+            if lambda == 0:
```

Comment:
You haven't given the forward in any of the other docstrings ;) But I guess it's in the user guide.